modules/apps/asset/asset-browser-web/src/main/resources/META-INF/resources/view.jsp

```diff
@@ -283,14 +285,14 @@
 					Liferay.Util.getOpener().Liferay.fire(
 						'<%= HtmlUtil.escapeJS(assetBrowserDisplayContext.getEventName()) %>',
 						{
-							data: event.delegateTarget.dataset,
+							data: event.target.closest('.selector-button').dataset,
```

Comment:
just to understand this, don't we have a way to get the `delegateTarget` so we don't need to do this `closest` stuff? Looks a bit weird to me at first sight

Reply:
AFAIK no, we don't have the `delegateTarget` functionality in vanilla JS, or in our `delegate` util.

Reply:
Yeah, I myself have seen this trying to update `senna` to use this delegate. I think we should be able to provide this directly in `delegate.es.js` to avoid having to always find `.closest` since the selector is already available when calling `delegate`.

Reply:
Only thing is it might not be exactly equivalent, because `closest` will find the closest selector at the time the event is fired, whereas anything we do instide `delegate()` will get configured at the time the subscription is created. Unlikely to be an issue in practice, but it is a distinction. I don't know what the old one used to do.

Reply:
> Only thing is it might not be exactly equivalent, because closest will find the closest selector at the time the event is fired, whereas anything we do inside delegate() will get configured at the time the subscription is created. Unlikely to be an issue in practice, but it is a distinction. I don't know what the old one used to do.

That depends, since you can do it also at the time the callback is called rather than at subscription time

Reply:
Although I'm not sure what to make of the [current implementation](https://github.com/metal/metal.js/blob/466812085dd29e37c90b835c64cb49555d76a294/packages/metal-dom/src/domNamed.js#L747)

Reply:
I'm currently playing around with the implementation where I add a `delegateTarget` property to the `event` object, but I guess that's what @wincent you said was dangerous?

The other avenue could be, like @jbalsas said earlier, to add a argument in the callback: 
```js
callback(event, delegateTarget = event.target.closest(selector)
```

IMO the 2nd approach might work, and it might be more safe. I'm just not sure if the default value for the argument will work in JSPs, haven't tested it yet.

Reply:
> but I guess that's what @wincent you said was dangerous?

Not dangerous per se, but possibly different from what the old one did. Seeing as the DOM can change between subscription and firing, it matters whether you derive the delegate at subscription-time or at fire-time. Based on what @jbalsas just link, it looks like the old one was walking up at fire-time, so if you're going to go down this path you should do the same (re-compute the delegate target every time an event fires, rather than caching it at subscription time).